### DISCUSSION TITLE
Graceful shutdown support with mongodb connection

### DISCUSSION THREAD

**Discussion by ruslankonev** [nuxt/nuxt#20457] [Questions]
What is the best way to close the connection to mongodb when updating (restarting) the application? 

Using next code in Nitro plugin be the right solution?

```ts
process.on('SIGTERM', async () => {
      logger.info('SIGTERM signal received.');
      await client.close();
});
```

**Comment by maou-shonen** [comment]
Although I haven't tried it,
I think the functionality you want can be implemented in the `close hook` of Nitro.
